entity-framework/core/managing-schemas/migrations/applying.md

```diff
@@ -0,0 +1,193 @@
+* SQL scripts can be provided to a DBA, and can be managed and archived separately.
+
+### [.NET Core CLI](#tab/dotnet-core-cli)
+
+#### Basic Usage
+
+The following generates a SQL script from a blank database to the latest migration:
+
+```dotnetcli
+dotnet ef migrations script
+```
+
+#### With From (to implied)
+
+The following generates a SQL script from the given migration to the latest migration.
+
+```dotnetcli
+dotnet ef migrations script AddNewTables
+```
+
+#### With From and To
+
+The following generates a SQL script from the specified `from` migration to the specified `to` migration.
+
+```dotnetcli
+dotnet ef migrations script AddNewTables AddAuditTable
+```
+
+You can use a `from` that is newer than the `to` in order to generate a rollback script.
+
+> [!WARNING]
+> Please take note of potential data loss scenarios.
+
+### [Visual Studio](#tab/vs)
+
+#### Basic Usage
+
+The following generates a SQL script from a blank database to the latest migration:
+
+``` powershell
+Script-Migration
+```
+
+#### With From (to implied)
+
+The following generates a SQL script from the given migration to the latest migration.
+
+```powershell
+Script-Migration AddNewTables
+```
+
+#### With From and To
+
+The following generates a SQL script from the specified `from` migration to the specified `to` migration.
+
+```powershell
+Script-Migration AddNewTables AddAuditTable
+```
+You can use a `from` that is newer than the `to` in order to generate a rollback script. *Please take note of potential data loss scenarios.*
+
+***
+
+Script generation accepts the following two arguments in indicate which range of migrations should be generated:
+
+* The **from** migration should be the last migration applied to the database before running the script. If no migrations have been applied, specify `0` (this is the default).
+* The **to** migration is the last migration that will be applied to the database after running the script. This defaults to the last migration in your project.
+
+## Idempotent SQL scripts
+
+The SQL scripts generated above can only be applied to change your schema from one migration to another; it is your responsibility to apply the script appropriately, and only to database in the correct migration state. EF Core also supports generating **idempotent** scripts, which internally check which migrations have already been applied (via the migrations history table), and only apply missing ones. This is useful if you don't exactly know what the last migration applied to the database was, or if you are deploying to multiple databases that may each be at a different migration.
+
+The following generates idempotent migrations:
+
+#### [.NET Core CLI](#tab/dotnet-core-cli)
+
+```dotnetcli
+dotnet ef migrations add AddNewTables --idempotent
+```
+
+#### [Visual Studio](#tab/vs)
+
+``` powershell
+Add-Migration AddNewTables -Idempotent
+```
+
+> [!WARNING]
+> Idempotent scripts have some known limitations and will not work in all scenarios. 
```

Comment:
Should we list some examples of the limitations here?

Reply:
Yeah, @bricelam commented on that - I'll touch up this section.

Reply:
Looking at this again, I ended up just removing the warning... As @bricelam said, most people probably won't run into these and I don't believe we should put every possible open issue in our conceptual docs...